widgetssdk/src/main/java/com/glia/widgets/helper/ViewExtensions.kt

```diff
@@ -145,3 +150,86 @@ internal val View.layoutInflater: LayoutInflater get() = LayoutInflater.from(thi
     TextViewCompat.setCompoundDrawableTintList(this, tint)
 }
+
+internal fun TextView.setLocaleText(@StringRes stringKey: Int, vararg values: StringKeyPair) {
+    registerLocaleListener(stringKey, *values) { upToDateTranslation ->
+        text = upToDateTranslation
+    }
+}
+
+internal fun TextView.setText(locale: LocaleString?) {
+    if (locale == null) {
+        text = ""
+        return
+    }
+    registerLocaleListener(locale.stringKey, *locale.values.toTypedArray()) { upToDateTranslation ->
+        text = upToDateTranslation
+    }
+}
+
+internal fun TextView.setLocaleHint(@StringRes stringKey: Int, vararg values: StringKeyPair) {
+    registerLocaleListener(stringKey, *values) { upToDateTranslation ->
+        hint = upToDateTranslation
+    }
+}
+
+internal fun TextView.setHint(locale: LocaleString?) {
+    if (locale == null) {
+        hint = ""
+        return
+    }
+    registerLocaleListener(locale.stringKey, *locale.values.toTypedArray()) { upToDateTranslation ->
+        hint = upToDateTranslation
+    }
+}
+
+internal fun View.setLocaleContentDescription(@StringRes stringKey: Int, vararg values: StringKeyPair) {
+    registerLocaleListener(stringKey, *values) { upToDateTranslation ->
+        contentDescription = upToDateTranslation
+    }
+}
+
+internal fun View.setContentDescription(locale: LocaleString?) {
+    if (locale == null) {
+        contentDescription = ""
+        return
+    }
+    registerLocaleListener(locale.stringKey, *locale.values.toTypedArray()) { upToDateTranslation ->
+        contentDescription = upToDateTranslation
+    }
+}
+
+internal fun Toolbar.setLocaleNavigationContentDescription(@StringRes stringKey: Int, vararg values: StringKeyPair) {
+    registerLocaleListener(stringKey, *values) { upToDateTranslation ->
+        navigationContentDescription = upToDateTranslation
+    }
+}
+
+private fun View.registerLocaleListener(@StringRes stringKey: Int, vararg values: StringKeyPair, listener: (String) -> Unit) {
+    val localeManager = Dependencies.getLocaleProvider()
+    val localeListener = {
+        // It is strange that it should already be UI thread according to:
+        // https://github.com/salemove/android-sdk/blob/4dbf361ecf108b478b997e415734240778af2e41/androidSdk/src/main/java/com/glia/androidsdk/internal/locale/LocaleManagerInternal.kt#L71
+        // But without `runOnUiThread` it sometimes does not work
+        context.asActivity()?.runOnUiThread {
+            listener.invoke(localeManager.getString(stringKey, values.toList()))
+        }
+    }
+    localeListener.invoke()
+    val disposable = localeManager.getLocaleObservable().subscribe(
+        { localeListener.invoke() },
+        { /* no-op */ }
+    )
+
+    addOnAttachStateChangeListener(object : View.OnAttachStateChangeListener {
+        override fun onViewAttachedToWindow(v: View) {
+            // TODO: to add listener back here or not to add, that is the question
```

Comment:
Is there a case when the same view can be re-attached to Window?

Reply:
Yes, but I don't know how common it is. It's probably a corner case. I remember (from very long ago) that it can happen when using multiple `Fragment`s.